src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpChangeNamespaceService.cs

```diff
@@ -470,16 +443,25 @@ private static ExpressionSyntax CreateNamespaceAsMemberAccess(ImmutableArray<str
         /// trivia around closing brace are concatenated together respectively.
         /// </summary>
         private static (ImmutableArray<SyntaxTrivia> openingTrivia, ImmutableArray<SyntaxTrivia> closingTrivia)
-            GetOpeningAndClosingTriviaOfNamespaceDeclaration(NamespaceDeclarationSyntax namespaceDeclaration)
+            GetOpeningAndClosingTriviaOfNamespaceDeclaration(BaseNamespaceDeclarationSyntax baseNamespace)
         {
             var openingBuilder = ArrayBuilder<SyntaxTrivia>.GetInstance();
-            openingBuilder.AddRange(namespaceDeclaration.GetLeadingTrivia());
-            openingBuilder.AddRange(namespaceDeclaration.OpenBraceToken.LeadingTrivia);
-            openingBuilder.AddRange(namespaceDeclaration.OpenBraceToken.TrailingTrivia);
-
             var closingBuilder = ArrayBuilder<SyntaxTrivia>.GetInstance();
-            closingBuilder.AddRange(namespaceDeclaration.CloseBraceToken.LeadingTrivia);
-            closingBuilder.AddRange(namespaceDeclaration.CloseBraceToken.TrailingTrivia);
+
+            openingBuilder.AddRange(baseNamespace.GetLeadingTrivia());
+
+            if (baseNamespace is NamespaceDeclarationSyntax namespaceDeclaration)
+            {
+                openingBuilder.AddRange(namespaceDeclaration.OpenBraceToken.LeadingTrivia);
+                openingBuilder.AddRange(namespaceDeclaration.OpenBraceToken.TrailingTrivia);
+
+                closingBuilder.AddRange(namespaceDeclaration.CloseBraceToken.LeadingTrivia);
+                closingBuilder.AddRange(namespaceDeclaration.CloseBraceToken.TrailingTrivia);
+            }
+            else if (baseNamespace is FileScopedNamespaceDeclarationSyntax fileScopedNamespace)
+            {
+                openingBuilder.AddRange(fileScopedNamespace.SemicolonToken.TrailingTrivia);
```

Comment:
Can the semicolon have leading trivia here?

Reply:
yes, though that's a 1:10000 case :)  i'm writing for the 99.9% cases :)

Reply:
@CyrusNajmabadi To me personally, I agree this case is very very rare. But since it's a simple change, I'd handle it. Users put comments in weird places 😄
I'd expect someone later reporting the following trivia being lost:

```
namespace    R
    /*trivia*/;
```

Reply:
to me it's that i can map where i expect teh trailing trivia to go.  Leading trivia is something else entirely.